server/test/unit/vitalsTest/checkForInternalProblemsTest.js

```diff
@@ -0,0 +1,215 @@
+      for (let i = 1; i <= maxStillnessAlerts + 1; i++) {
+        this[`session${i}`] = await sessionDBFactory(db, {
+          locationid: this.testLocation.locationid,
+          alertType: this.alertType,
+          incidentCategory: this.incidentCategory,
+        })
+        await helpers.sleep(1) // sleep to make sure that these have created_at values at least one millisecond apart
+      }
+      await vitals.checkForInternalProblems()
+    })
+
+    it('should send one max stillness alert to Sentry', () => {
+      expect(helpers.logSentry).to.have.been.calledOnce
+    })
+
+    it('should send one max stillness alert specific to the location that has exceeded the maxStillnessAlerts in 24 hours to Sentry', () => {
+      expect(helpers.logSentry).to.have.been.calledOnceWithExactly(
+        `Unusually frequent number of stillness alerts have been received at ${this.testLocation.locationid}`,
+      )
+    })
+
+    it('should not log any errors', () => {
+      expect(helpers.log).to.not.be.called
+    })
+  })
+
+  describe('should log to Sentry for more than maxStillnessAlerts, ignores present duration alerts in the past 24 hours for a location', () => {
+    beforeEach(async () => {
+      // Insert a single location and maxStillnessAlerts sessions + 1
+      const client = await factories.clientDBFactory(db, { isActive: true })
+      this.testLocation = await locationDBFactory(db, {
+        clientId: client.id,
+        isActive: true,
+      })
+      for (let i = 1; i <= maxStillnessAlerts + 1; i++) {
+        this[`session${i}`] = await sessionDBFactory(db, {
+          locationid: this.testLocation.locationid,
+          alertType: ALERT_TYPE.SENSOR_DURATION,
+          incidentCategory: ALERT_TYPE.SENSOR_DURATION,
+        })
+        await helpers.sleep(1) // sleep to make sure that these have created_at values at least one millisecond apart
+      }
+      for (let i = maxStillnessAlerts + 2; i <= (maxStillnessAlerts + 2) * 2; i++) {
+        this[`session${i}`] = await sessionDBFactory(db, {
+          locationid: this.testLocation.locationid,
+          alertType: ALERT_TYPE.SENSOR_STILLNESS,
+          incidentCategory: ALERT_TYPE.SENSOR_STILLNESS,
+        })
+        await helpers.sleep(1) // sleep to make sure that these have created_at values at least one millisecond apart
+      }
+      await vitals.checkForInternalProblems()
+    })
+
+    it('should send one max stillness alert to Sentry', () => {
+      expect(helpers.logSentry).to.have.been.calledOnce
+    })
+
+    it('should send one max stillness alert specific to the location that has exceeded the maxStillnessAlerts in 24 hours to Sentry', () => {
+      expect(helpers.logSentry).to.have.been.calledOnceWithExactly(
+        `Unusually frequent number of stillness alerts have been received at ${this.testLocation.locationid}`,
+      )
+    })
+
+    it('should not log any errors', () => {
+      expect(helpers.log).to.not.be.called
+    })
+  })
+})
```

Comment:
I'd also add test(s) around when there are Stillness Alerts outside the 24 hours.

Reply:
Really good point, thank you! I'll get to making those tests : )